packages/nextjs/src/intl/languages.ts

```diff
@@ -21,5 +21,19 @@ export function getLanguageFromNextLocaleIsoCode<
     }
 
     const locale = LocaleObject.fromAnyCode(nextLocaleIsoCode);
+    const shortRegionCode = locale.toRegionCode();
+    const neutralLanguageCode = locale.toNeutralLanguageCode();
+
+    const languageWithSameRegionAndNeutralCode = languages.find(({ code }) => {
+        const comparingLocale = LocaleObject.fromAnyCode(code);
+        const hasSameNeutralCode = comparingLocale.toNeutralLanguageCode() === neutralLanguageCode;
+        const hasSameRegionCode = comparingLocale.toRegionCode() === shortRegionCode;
+        return hasSameNeutralCode && hasSameRegionCode;
+    });
+
+    if (languageWithSameRegionAndNeutralCode) {
+        return languageWithSameRegionAndNeutralCode;
+    }
+
```

Comment:
I know you're tired of this already, but I was meaning to put this logic inside `getLanguageFromLocaleIsoCode` :D

Reply:
Ugh, my bad. Sorry :D

Btw, where do you think is the best place for the logic to be? I believe right after this check, right?

https://github.com/prezly/theme-kit-js/blob/01337801b3e5728700b798b02c47e9d7c9486175/packages/core/src/intl/languages.ts#L228

Reply:
Exactly!